### PR TITLE
Extend by the Code Climate output format

### PR DESCRIPTION
**Pull request by lukas9393**
Hi, we use this linter in our PHP projects. We have our repositories in GitLab and use the CI pipeline there. When we run the CI pipeline, we run QA tools. When the pipeline is executed, we want GitLab to load the errors from a json file and display them in the merge request. This should make it easier to see which bugs have been added and which have been fixed.

For this we use the [Code Quality](https://docs.gitlab.com/ee/user/project/merge_requests/code_quality.html#implementing-a-custom-tool) widget. This requires creating a Json file that implements a subset of the code-climate specification. 

With this merge request I suggest that this linter supports this format so that not every CI pipeline has to use a converter.

We also use PHPStan, they support this [format](https://github.com/phpstan/phpstan/blob/master/website/src/user-guide/output-format.md). 

PHP_CodeSniffer do not support the format, but they offer the possibility to include external [reports/formats](https://github.com/lukas9393/PHP_CodeSniffer/commit/77f77c69c42af17a564fedf707937fef32a754d4). I have not discovered this possibility in this linter. We strongly want to avoid using a fork for this feature, so I made this merge request.

Sorry for bothering you with this issue.

### CODE REVIEW

**Comment by lukas9393** [comment]
Is there any information missing or does the pull request not match your requirements?

**Comment by grogy** [comment]
For the first look it is ok. I will try on any project and is possible to merge it

File: src/Output.php [hunk 1]
```diff
@@ -85,6 +85,78 @@ public function writeResult(Result $result, ErrorFormatter $errorFormatter, $ign
     }
 }
 
+class GitLabOutput implements Output
```
Comment (roelofr) [inline src/Output.php:88]: It might be a good idea to merge this after the PSR-4 changes have been pushed through. Otherwise, the `src/Output.php` file keeps growing. See #45.

File: tests/Settings.parseArguments.phpt [hunk 2]
```diff
@@ -103,6 +103,14 @@ class SettingsParseArgumentsTest extends Tester\TestCase
         Assert::equal(Settings::FORMAT_JSON, $settings->format);
     }
 
+    public function testGitLabOutput()
```
Comment (roelofr) [inline tests/Settings.parseArguments.phpt:106]: This test only checks if the `--gitlab` argument works. You should also test if the JSON output is valid according to [GitLab's specifications][1] (or maybe even [CodeClimate's standards][2]).

[1]: https://docs.gitlab.com/ee/user/project/merge_requests/code_quality.html#implementing-a-custom-tool
[2]: https://github.com/codeclimate/platform/blob/master/spec/analyzers/SPEC.md#data-types

**Review by roelofr — CHANGES_REQUESTED** [review]
Looks good, but your tests could be a bit more thorough.

It might even be an idea to support CodeClimate directly, which would automatically support GitLab too.

I'd recommend waiting for, or rebasing onto #45.

File: src/Output.php (hunk 1 above)
Comment (grogy) [reply src/Output.php:88]: PR from the current state is ok. Is my fault that #45 is not done.

File: README.md [hunk 3]
```diff
@@ -51,19 +51,20 @@ This is a fork of [original project](https://github.com/JakubOnderka/PHP-Paralle
 - `-p <php>`        		Specify PHP-CGI executable to run (default: 'php').
 - `-s, --short`     		Set short_open_tag to On (default: Off).
 - `-a, --asp`       		Set asp_tags to On (default: Off).
-- `-e <ext>`        		Check only files with selected extensions separated by comma. (default: php,php3,php4,php5,phtml,phpt)
+- `-e <ext>`        		Check only files with selected extensions separated by commas (default: php,php3,php4,php5,phtml,phpt).
```
Comment (grogy) [inline README.md:54]: Can be in a separate commit.

File: tests/Settings.parseArguments.phpt (hunk 2 above)
Comment (grogy) [reply tests/Settings.parseArguments.phpt:106]: I agree. Please, can you add the unit test? I will be easy :)

**Review by grogy — CHANGES_REQUESTED** [review]

File: tests/Settings.parseArguments.phpt (hunk 2 above)
Comment (lukas9393) [reply tests/Settings.parseArguments.phpt:106]: I have added a test for this. I hope this is what you had in mind for your tests.

File: README.md (hunk 3 above)
Comment (lukas9393) [reply README.md]: I removed it from the PR.

**Comment by lukas9393** [comment]
Does the test `tests/OutputTest::testGitLabOutput` meet your expectations or is there anything that can be improved? 
Is the status that we are waiting for PR #45?

**Comment by grogy** [comment]
PR #45 need more time from me, so i merged this. Thank you :+1:
